### ISSUE TITLE
Regression Model Performance Evaluation

### ISSUE THREAD

**Issue by thehanggit** [cagov/caldata-mdsa-caltrans-pems#494]
The regression models are trained every 3 months using data from the 3rd day of the beginning of each season. Since imputed data lacks ground truth, we cannot directly evaluate the performance by comparing predictions on imputed data. Instead, the evaluation focuses on assessing how well the trained model generalizes to missing data.

Evaluation Steps:

1. EDA analysis on the training data to examine whether distribution, correlation, and normality assumption are satisfied.
2. Choose one month of data within a season and randomly drop a subset of ground truth data (observed data) to create a test set.
3. Use the fixed coefficients from trained model to make predictions for the test set.
4. Compare predictions against the actual values in the test set using MAE, RMSE, and R^2.
5. Repeat the random sampling and evaluation through k-fold cross-validation to avoid bias.
6. Compare the results of different regression models.

**Comment by ian-r-rose** [comment]
Love this!

**Comment by thehanggit** [comment]
> Love this!

Thank you! @ian-r-rose. Will see what happens

**Comment by ian-r-rose** [comment]
I'm still digesting this, but I love the analysis @thehanggit!

> In summary, throughout the analysis, the findings are consistent among different evaluation methods and segmentation. GLOBAL imputation methods provide more accurate imputation results than REGION and LOCAL. But I noticed that in the assignment, we chose the order of LOCAL, REGIONAL, and GLOBAL. I suggest we should reverse it.

I'm really happy to see you suggest this @thehanggit! We initially chose LOCAL > REGIONAL > GLOBAL for two reasons:

1. Existing PeMS prefers local to global
2. It makes a certain amount of sense that nearby stations would have more information to provide when imputing that ones further away. If there is some super local (like a fender bender or a mattress on the freeway), we might expect closer stations to provide a more accurate estimate.

That said, another thing we have seen is that these data are pretty noisy, even for stations that are considered "good". Imputing from one or a few neighboring stations tends to replicate that noise. But in the global estimate we are using many more stations as an input, and can beat down the noise a bit. At least, that's my interpretation of what's going on. When I was doing some visualization of the regressions last year I was struck by how much more stable the global one seemed, so it's nice to see that confirmed. 

One question that wasn't clear to me from your analysis: do you have a sense of how good these imputed values are in a _relative_ sense? That is to say, something like the MAE compared to the average value (so, on average x% off). That would help me build some intuition on whether any of these imputation techniques are any good, or if they are all bad.
